# Open pangenome of *Lactococcus lactis*
In our study, we built a high-quality genome catalog of *Lactococcus lactis* combining isolate genomes and metagenome-assembled genomes (MAGs). Our approach to reconstructing genomes from the food metagenomes exploited metagenomic single-sample assembly, contig binning, and taxonomic annotation of genomes. Isolate genomes and human MAGs were downloaded from publicly available databases. After integrating these MAGs and isolate genomes of *L. lactis* together, genomes were then conducted strictly quality control to identity high-quality genomes (completeness >90% and contamination <5%).



<div align="center">
    <img src="https://github.com/skye-zhai/analysis-pipeline/blob/main/Pipeline.png" height="600" width="400">
</div>





<div align="center">



The comprehensive genome catalog was then used for various downstream analysis, like pangenome analysis, phylogenetic analysis, ANI calculation, functional enrichment analysis and so on. In the end, we generated a pangenome of *L. lactis* which was still open. These various analyses shed further light on unexpectedly high diversity within the taxon at both genome and gene levels and gave clues about its population structure and evolution. 



The following resources are provided:

1. Scripts used in the analysis process are displayed. The parameters of various bioinformatics tools are also mentioned in the manuscript.
2. Some analysis results, including the resulting pangenome sequences, gene presence and absence tables and functional information.
3. Bulk download for the 163 high-quality genomes of *L. lactis* is available at https://cgm.sjtu.edu.cn/Lactococcus_lactis-pangenome/genome_catalog_of_Lactococcus_lactis.zip.
